DevTools/UserspaceEmulator/Emulator.cpp

```diff
@@ -492,6 +494,12 @@ int Emulator::virt$dbgputstr(FlatPtr characters, int length)
     return 0;
 }
 
+int Emulator::virt$chmod(FlatPtr path_addr, size_t path_length, mode_t mode)
+{
+    auto path = String::copy(mmu().copy_buffer_from_vm(path_addr, path_length));
```

Comment:
I don't think you need to make a `String` here as the syscall takes characters+length.

Reply:
Hmmm...I copied it to a string because I saw `realpath` and few other places doing the same, but now that you say it, I don't think any of them need this except maybe `execve`
Should I also replace the rest?

Reply:
Sure that could be cool :)